client/components/ImageViewer.vue

```diff
@@ -78,8 +105,34 @@ export default {
 			}
 
 			this.$root.$off("resize", this.correctPosition);
+			this.channel = null;
+			this.previousImage = null;
+			this.nextImage = null;
 			this.link = null;
 		},
+		getNewImage(delta) {
```

Comment:
Better off assigning next/prev within this once, so it doesn't have to filter twice.